{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_JCP2022</th>\n",
       "      <th>Metadata_broad_sample</th>\n",
       "      <th>Metadata_Name</th>\n",
       "      <th>Metadata_Vector</th>\n",
       "      <th>Metadata_Transcript</th>\n",
       "      <th>Metadata_Symbol</th>\n",
       "      <th>Metadata_NCBI_Gene_ID</th>\n",
       "      <th>Metadata_Taxon_ID</th>\n",
       "      <th>Metadata_Gene_Description</th>\n",
       "      <th>Metadata_Prot_Match</th>\n",
       "      <th>Metadata_Insert_Length</th>\n",
       "      <th>Metadata_pert_type</th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JCP2022_900001</td>\n",
       "      <td>BAD CONSTRUCT</td>\n",
       "      <td>BAD CONSTRUCT</td>\n",
       "      <td>pLX_304</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nan</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>trt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JCP2022_900002</td>\n",
       "      <td>ccsbBroad304_00001</td>\n",
       "      <td>ORF008415.1_TRC304.1</td>\n",
       "      <td>pLX_304</td>\n",
       "      <td>NM_001160173.3</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>9</td>\n",
       "      <td>9606</td>\n",
       "      <td>N-acetyltransferase 1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>870.0</td>\n",
       "      <td>trt</td>\n",
       "      <td>NAT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JCP2022_900003</td>\n",
       "      <td>ccsbBroad304_00002</td>\n",
       "      <td>ORF005388.1_TRC304.1</td>\n",
       "      <td>pLX_304</td>\n",
       "      <td>NM_001088.3</td>\n",
       "      <td>AANAT</td>\n",
       "      <td>15</td>\n",
       "      <td>9606</td>\n",
       "      <td>aralkylamine N-acetyltransferase</td>\n",
       "      <td>100.0</td>\n",
       "      <td>621.0</td>\n",
       "      <td>trt</td>\n",
       "      <td>AANAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JCP2022_900004</td>\n",
       "      <td>ccsbBroad304_00003</td>\n",
       "      <td>ORF003876.1_TRC304.1</td>\n",
       "      <td>pLX_304</td>\n",
       "      <td>NM_000663.4</td>\n",
       "      <td>ABAT</td>\n",
       "      <td>18</td>\n",
       "      <td>9606</td>\n",
       "      <td>4-aminobutyrate aminotransferase</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1500.0</td>\n",
       "      <td>trt</td>\n",
       "      <td>ABAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JCP2022_900005</td>\n",
       "      <td>ccsbBroad304_00007</td>\n",
       "      <td>ORF004679.1_TRC304.1</td>\n",
       "      <td>pLX_304</td>\n",
       "      <td>NM_000018.4</td>\n",
       "      <td>ACADVL</td>\n",
       "      <td>37</td>\n",
       "      <td>9606</td>\n",
       "      <td>acyl-CoA dehydrogenase very long chain</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1965.0</td>\n",
       "      <td>trt</td>\n",
       "      <td>ACADVL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_JCP2022 Metadata_broad_sample         Metadata_Name  \\\n",
       "0   JCP2022_900001         BAD CONSTRUCT         BAD CONSTRUCT   \n",
       "1   JCP2022_900002    ccsbBroad304_00001  ORF008415.1_TRC304.1   \n",
       "2   JCP2022_900003    ccsbBroad304_00002  ORF005388.1_TRC304.1   \n",
       "3   JCP2022_900004    ccsbBroad304_00003  ORF003876.1_TRC304.1   \n",
       "4   JCP2022_900005    ccsbBroad304_00007  ORF004679.1_TRC304.1   \n",
       "\n",
       "  Metadata_Vector Metadata_Transcript Metadata_Symbol Metadata_NCBI_Gene_ID  \\\n",
       "0         pLX_304                 NaN             NaN                   nan   \n",
       "1         pLX_304      NM_001160173.3            NAT1                     9   \n",
       "2         pLX_304         NM_001088.3           AANAT                    15   \n",
       "3         pLX_304         NM_000663.4            ABAT                    18   \n",
       "4         pLX_304         NM_000018.4          ACADVL                    37   \n",
       "\n",
       "  Metadata_Taxon_ID               Metadata_Gene_Description  \\\n",
       "0               NaN                                     NaN   \n",
       "1              9606                   N-acetyltransferase 1   \n",
       "2              9606        aralkylamine N-acetyltransferase   \n",
       "3              9606        4-aminobutyrate aminotransferase   \n",
       "4              9606  acyl-CoA dehydrogenase very long chain   \n",
       "\n",
       "  Metadata_Prot_Match Metadata_Insert_Length Metadata_pert_type  \\\n",
       "0                 NaN                    NaN                trt   \n",
       "1               100.0                  870.0                trt   \n",
       "2               100.0                  621.0                trt   \n",
       "3               100.0                 1500.0                trt   \n",
       "4               100.0                 1965.0                trt   \n",
       "\n",
       "  Metadata_Approved_Symbol  \n",
       "0                      NaN  \n",
       "1                     NAT1  \n",
       "2                    AANAT  \n",
       "3                     ABAT  \n",
       "4                   ACADVL  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_df = pd.read_csv('output/orf.csv.gz', dtype=str)\n",
    "orf_df[\"Metadata_NCBI_Gene_ID\"] = orf_df[\"Metadata_NCBI_Gene_ID\"].astype(str)\n",
    "\n",
    "orf_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_JCP2022</th>\n",
       "      <th>Metadata_NCBI_Gene_ID</th>\n",
       "      <th>Metadata_Symbol</th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JCP2022_800001</td>\n",
       "      <td>nan</td>\n",
       "      <td>no-guide</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JCP2022_800002</td>\n",
       "      <td>nan</td>\n",
       "      <td>non-targeting</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JCP2022_800003</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JCP2022_800004</td>\n",
       "      <td>127550</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>A3GALT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JCP2022_800005</td>\n",
       "      <td>53947</td>\n",
       "      <td>A4GALT</td>\n",
       "      <td>A4GALT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_JCP2022 Metadata_NCBI_Gene_ID Metadata_Symbol  \\\n",
       "0   JCP2022_800001                   nan        no-guide   \n",
       "1   JCP2022_800002                   nan   non-targeting   \n",
       "2   JCP2022_800003                     2             A2M   \n",
       "3   JCP2022_800004                127550         A3GALT2   \n",
       "4   JCP2022_800005                 53947          A4GALT   \n",
       "\n",
       "  Metadata_Approved_Symbol  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      A2M  \n",
       "3                  A3GALT2  \n",
       "4                   A4GALT  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crispr_df = pd.read_csv('output/crispr.csv.gz', dtype=str)\n",
    "crispr_df[\"Metadata_NCBI_Gene_ID\"] = crispr_df[\"Metadata_NCBI_Gene_ID\"].astype(str)\n",
    "crispr_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read metadata files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_complexname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M</td>\n",
       "      <td>A2M-BCHE complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>Nuclear pore complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AATF</td>\n",
       "      <td>AATF-NGDN-NOL10 complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCB1</td>\n",
       "      <td>ABCB1-ANXA2-RACK1-SRC complex|ABCB1-PPP2R3C-TF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCB11</td>\n",
       "      <td>ABCB11-CHMP5 complex</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol                               Metadata_complexname\n",
       "0                      A2M                                   A2M-BCHE complex\n",
       "1                     AAAS                               Nuclear pore complex\n",
       "2                     AATF                            AATF-NGDN-NOL10 complex\n",
       "3                    ABCB1  ABCB1-ANXA2-RACK1-SRC complex|ABCB1-PPP2R3C-TF...\n",
       "4                   ABCB11                               ABCB11-CHMP5 complex"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corum_df = pd.read_csv(\"output/CORUM_proteincomplex_annotations.tsv\", sep=\"\\t\").rename(\n",
    "    columns={\n",
    "        \"Symbol\": \"Metadata_Approved_Symbol\",\n",
    "        \"ComplexName\": \"Metadata_complexname\",\n",
    "    }\n",
    ")\n",
    "corum_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_protein_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>Plasma proteins|Predicted intracellular protei...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>Predicted intracellular proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>Cancer-related genes|Candidate cardiovascular ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>Disease related genes|Predicted intracellular ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>Enzymes|Predicted membrane proteins</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol                             Metadata_protein_class\n",
       "0                     A1BG  Plasma proteins|Predicted intracellular protei...\n",
       "1                     A1CF                   Predicted intracellular proteins\n",
       "2                      A2M  Cancer-related genes|Candidate cardiovascular ...\n",
       "3                    A2ML1  Disease related genes|Predicted intracellular ...\n",
       "4                  A3GALT2                Enzymes|Predicted membrane proteins"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "class_df = pd.read_csv(\n",
    "    \"output/human_protein_atlas_class_annotations.tsv.gz\", sep=\"\\t\"\n",
    ").rename(\n",
    "    columns={\n",
    "        \"Gene\": \"Metadata_Approved_Symbol\",\n",
    "        \"protein_class\": \"Metadata_protein_class\",\n",
    "    }\n",
    ")\n",
    "class_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_disease_involvement</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M</td>\n",
       "      <td>Cancer-related genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>Disease variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>Palmoplantar keratoderma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AANAT</td>\n",
       "      <td>Disease variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS1</td>\n",
       "      <td>Charcot-Marie-Tooth disease|Disease variant|Ep...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol                       Metadata_disease_involvement\n",
       "0                      A2M                               Cancer-related genes\n",
       "1                     AAAS                                    Disease variant\n",
       "2                    AAGAB                           Palmoplantar keratoderma\n",
       "3                    AANAT                                    Disease variant\n",
       "4                    AARS1  Charcot-Marie-Tooth disease|Disease variant|Ep..."
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disease_df = pd.read_csv(\n",
    "    \"output/human_protein_atlas_disease_annotations.tsv.gz\", sep=\"\\t\"\n",
    ").rename(\n",
    "    columns={\n",
    "        \"Gene\": \"Metadata_Approved_Symbol\",\n",
    "        \"disease_involvement\": \"Metadata_disease_involvement\",\n",
    "    }\n",
    ")\n",
    "disease_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_subcellular_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>Nucleoplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>Mitochondria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>Nuclear membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AACS</td>\n",
       "      <td>Vesicles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>Plasma membrane</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol Metadata_subcellular_location\n",
       "0                     A1CF                   Nucleoplasm\n",
       "1                   A4GALT                  Mitochondria\n",
       "2                     AAAS              Nuclear membrane\n",
       "3                     AACS                      Vesicles\n",
       "4                    AADAT               Plasma membrane"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "location_df = pd.read_csv(\n",
    "    \"output/human_protein_atlas_location_annotations.tsv.gz\",\n",
    "    sep=\"\\t\",\n",
    "    usecols=[\"Gene\", \"subcellular_location\"],\n",
    ").rename(\n",
    "    columns={\n",
    "        \"Gene\": \"Metadata_Approved_Symbol\",\n",
    "        \"subcellular_location\": \"Metadata_subcellular_location\",\n",
    "    }\n",
    ")\n",
    "location_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_gene_group_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A12M1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A12M2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A12M3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A12M4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>594</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol Metadata_gene_group_id\n",
       "0                    A12M1                    NaN\n",
       "1                    A12M2                    NaN\n",
       "2                    A12M3                    NaN\n",
       "3                    A12M4                    NaN\n",
       "4                     A1BG                    594"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_family = pd.read_csv(\n",
    "    \"output/hgnc_approved_symbol_processed.tsv\",\n",
    "    sep=\"\\t\",\n",
    "    usecols=[\"Approved_symbol\", \"Gene_group_ID\"],\n",
    ").rename(\n",
    "    columns={\n",
    "        \"Approved_symbol\": \"Metadata_Approved_Symbol\",\n",
    "        \"Gene_group_ID\": \"Metadata_gene_group_id\",\n",
    "    }\n",
    ")\n",
    "gene_family.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_homologue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol  Metadata_homologue\n",
       "0                     A1BG                True\n",
       "1                     A1CF                True\n",
       "2                      A2M                True\n",
       "3                    A2ML1                True\n",
       "4                  A3GALT2                True"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "homolog_df = pd.read_csv(\"output/homologues.tsv\", sep=\"\\t\").rename(\n",
    "    columns={\"Symbol\": \"Metadata_Approved_Symbol\", \"Homologue\": \"Metadata_homologue\"}\n",
    ")\n",
    "\n",
    "homolog_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_biological_process</th>\n",
       "      <th>Metadata_cellular_component</th>\n",
       "      <th>Metadata_molecular_function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>GOBP_BASE_CONVERSION_OR_SUBSTITUTION_EDITING|G...</td>\n",
       "      <td>GOCC_APOLIPOPROTEIN_B_MRNA_EDITING_ENZYME_COMP...</td>\n",
       "      <td>GOMF_DOUBLE_STRANDED_RNA_BINDING|GOMF_MRNA_BIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>GOBP_ACTIVATION_OF_IMMUNE_RESPONSE|GOBP_ACUTE_...</td>\n",
       "      <td>GOCC_BLOOD_MICROPARTICLE|GOCC_COLLAGEN_CONTAIN...</td>\n",
       "      <td>GOMF_CALCIUM_DEPENDENT_PROTEIN_BINDING|GOMF_CH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>GOBP_NEGATIVE_REGULATION_OF_CATALYTIC_ACTIVITY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GOMF_ENDOPEPTIDASE_REGULATOR_ACTIVITY|GOMF_ENZ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>GOBP_CARBOHYDRATE_DERIVATIVE_BIOSYNTHETIC_PROC...</td>\n",
       "      <td>GOCC_GOLGI_APPARATUS|GOCC_GOLGI_APPARATUS_SUBC...</td>\n",
       "      <td>GOMF_GALACTOSYLTRANSFERASE_ACTIVITY|GOMF_GLYCO...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>GOBP_CARBOHYDRATE_DERIVATIVE_BIOSYNTHETIC_PROC...</td>\n",
       "      <td>GOCC_GOLGI_APPARATUS|GOCC_GOLGI_MEMBRANE</td>\n",
       "      <td>GOMF_ACETYLGLUCOSAMINYLTRANSFERASE_ACTIVITY|GO...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol                        Metadata_biological_process  \\\n",
       "0                     A1CF  GOBP_BASE_CONVERSION_OR_SUBSTITUTION_EDITING|G...   \n",
       "1                      A2M  GOBP_ACTIVATION_OF_IMMUNE_RESPONSE|GOBP_ACUTE_...   \n",
       "2                    A2ML1  GOBP_NEGATIVE_REGULATION_OF_CATALYTIC_ACTIVITY...   \n",
       "3                  A3GALT2  GOBP_CARBOHYDRATE_DERIVATIVE_BIOSYNTHETIC_PROC...   \n",
       "4                   A4GALT  GOBP_CARBOHYDRATE_DERIVATIVE_BIOSYNTHETIC_PROC...   \n",
       "\n",
       "                         Metadata_cellular_component  \\\n",
       "0  GOCC_APOLIPOPROTEIN_B_MRNA_EDITING_ENZYME_COMP...   \n",
       "1  GOCC_BLOOD_MICROPARTICLE|GOCC_COLLAGEN_CONTAIN...   \n",
       "2                                                NaN   \n",
       "3  GOCC_GOLGI_APPARATUS|GOCC_GOLGI_APPARATUS_SUBC...   \n",
       "4           GOCC_GOLGI_APPARATUS|GOCC_GOLGI_MEMBRANE   \n",
       "\n",
       "                         Metadata_molecular_function  \n",
       "0  GOMF_DOUBLE_STRANDED_RNA_BINDING|GOMF_MRNA_BIN...  \n",
       "1  GOMF_CALCIUM_DEPENDENT_PROTEIN_BINDING|GOMF_CH...  \n",
       "2  GOMF_ENDOPEPTIDASE_REGULATOR_ACTIVITY|GOMF_ENZ...  \n",
       "3  GOMF_GALACTOSYLTRANSFERASE_ACTIVITY|GOMF_GLYCO...  \n",
       "4  GOMF_ACETYLGLUCOSAMINYLTRANSFERASE_ACTIVITY|GO...  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_df = pd.read_csv(\n",
    "    \"output/gene-ontology-annotations.tsv.gz\", sep=\"\\t\"\n",
    ").rename(\n",
    "    columns={\n",
    "        \"gene\": \"Metadata_Approved_Symbol\",\n",
    "        \"go_bp_list\": \"Metadata_biological_process\",\n",
    "        \"go_cc_list\": \"Metadata_cellular_component\",\n",
    "        \"go_mf_list\": \"Metadata_molecular_function\",\n",
    "    }\n",
    ")\n",
    "go_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_Approved_Symbol</th>\n",
       "      <th>Metadata_wikipathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>WP_GANGLIO_SPHINGOLIPID_METABOLISM|WP_GLYCOSPH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>WP_GLYCOSPHINGOLIPID_METABOLISM|WP_SYNTHESIS_O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>WP_PRADER_WILLI_AND_ANGELMAN_SYNDROME</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>WP_IDO_METABOLIC_PATHWAY|WP_KYNURENINE_PATHWAY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>WP_NUCLEOTIDE_BINDING_OLIGOMERIZATION_DOMAIN_N...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_Approved_Symbol                               Metadata_wikipathway\n",
       "0                  A3GALT2  WP_GANGLIO_SPHINGOLIPID_METABOLISM|WP_GLYCOSPH...\n",
       "1                   A4GALT  WP_GLYCOSPHINGOLIPID_METABOLISM|WP_SYNTHESIS_O...\n",
       "2                     AAAS              WP_PRADER_WILLI_AND_ANGELMAN_SYNDROME\n",
       "3                    AADAT  WP_IDO_METABOLIC_PATHWAY|WP_KYNURENINE_PATHWAY...\n",
       "4                     AAMP  WP_NUCLEOTIDE_BINDING_OLIGOMERIZATION_DOMAIN_N..."
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wp_df = pd.read_csv(\"output/wikipathway-annotations.tsv.gz\", sep=\"\\t\").rename(\n",
    "    columns={\n",
    "        \"gene\": \"Metadata_Approved_Symbol\",\n",
    "        \"wikipathway\": \"Metadata_wikipathway\",\n",
    "    }\n",
    ")\n",
    "wp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_NCBI_Gene_ID</th>\n",
       "      <th>Metadata_dependency_probability</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0.025671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>29974</td>\n",
       "      <td>0.006055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>0.014716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>144568</td>\n",
       "      <td>0.042013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>127550</td>\n",
       "      <td>0.040916</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_NCBI_Gene_ID  Metadata_dependency_probability\n",
       "0                     1                         0.025671\n",
       "1                 29974                         0.006055\n",
       "2                     2                         0.014716\n",
       "3                144568                         0.042013\n",
       "4                127550                         0.040916"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dependency_df = pd.read_csv(\n",
    "    \"output/CRISPRGeneDependency_U2OS.csv.gz\", usecols=[\"geneID\", \"probability\"]\n",
    ").rename(\n",
    "    columns={\n",
    "        \"geneID\": \"Metadata_NCBI_Gene_ID\",\n",
    "        \"probability\": \"Metadata_dependency_probability\",\n",
    "    }\n",
    ")\n",
    "\n",
    "dependency_df.Metadata_NCBI_Gene_ID = dependency_df.Metadata_NCBI_Gene_ID.astype(str)\n",
    "dependency_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_NCBI_Gene_ID</th>\n",
       "      <th>Metadata_TPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7105</td>\n",
       "      <td>4.519164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>64102</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8813</td>\n",
       "      <td>6.673698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>57147</td>\n",
       "      <td>2.134221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>55732</td>\n",
       "      <td>4.107688</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_NCBI_Gene_ID  Metadata_TPM\n",
       "0                  7105      4.519164\n",
       "1                 64102      0.000000\n",
       "2                  8813      6.673698\n",
       "3                 57147      2.134221\n",
       "4                 55732      4.107688"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_df = pd.read_csv(\n",
    "    \"output/OmicsExpressionProteinCodingGenesTPMLogp1_U2OS.csv.gz\",\n",
    "    usecols=[\"geneID\", \"TPM\"],\n",
    ").rename(columns={\"geneID\": \"Metadata_NCBI_Gene_ID\", \"TPM\": \"Metadata_TPM\"})\n",
    "\n",
    "expression_df.Metadata_NCBI_Gene_ID = expression_df.Metadata_NCBI_Gene_ID.astype(str)\n",
    "expression_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Metadata_NCBI_Gene_ID</th>\n",
       "      <th>Metadata_gene_effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>-0.129731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>29974</td>\n",
       "      <td>0.283889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>0.004612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>144568</td>\n",
       "      <td>-0.070108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>127550</td>\n",
       "      <td>-0.114891</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Metadata_NCBI_Gene_ID  Metadata_gene_effect\n",
       "0                     1             -0.129731\n",
       "1                 29974              0.283889\n",
       "2                     2              0.004612\n",
       "3                144568             -0.070108\n",
       "4                127550             -0.114891"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_effect_df = pd.read_csv(\n",
    "    \"output/CRISPRGeneEffect_U2OS.csv.gz\", usecols=[\"geneID\", \"effect\"]\n",
    ").rename(columns={\"geneID\": \"Metadata_NCBI_Gene_ID\", \"effect\": \"Metadata_gene_effect\"})\n",
    "\n",
    "gene_effect_df.Metadata_NCBI_Gene_ID = gene_effect_df.Metadata_NCBI_Gene_ID.astype(str)\n",
    "gene_effect_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Metadata_JCP2022                   15142\n",
       "Metadata_broad_sample              15142\n",
       "Metadata_Name                      15132\n",
       "Metadata_Vector                    15142\n",
       "Metadata_Transcript                15106\n",
       "Metadata_Symbol                    15107\n",
       "Metadata_NCBI_Gene_ID              15142\n",
       "Metadata_Taxon_ID                  15107\n",
       "Metadata_Gene_Description          14853\n",
       "Metadata_Prot_Match                14320\n",
       "Metadata_Insert_Length             15141\n",
       "Metadata_pert_type                 15142\n",
       "Metadata_Approved_Symbol           14923\n",
       "Metadata_complexname                3790\n",
       "Metadata_protein_class             14484\n",
       "Metadata_disease_involvement        4683\n",
       "Metadata_subcellular_location      10363\n",
       "Metadata_gene_group_id             11305\n",
       "Metadata_homologue                 13482\n",
       "Metadata_biological_process        12777\n",
       "Metadata_cellular_component        10464\n",
       "Metadata_molecular_function        11805\n",
       "Metadata_wikipathway                6630\n",
       "Metadata_dependency_probability    13922\n",
       "Metadata_TPM                       14491\n",
       "Metadata_gene_effect               13808\n",
       "dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_df = (\n",
    "    orf_df.merge(corum_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(class_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(disease_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(location_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(gene_family, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(homolog_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(go_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(wp_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(dependency_df, how=\"left\", on=\"Metadata_NCBI_Gene_ID\")\n",
    "    .merge(expression_df, how=\"left\", on=\"Metadata_NCBI_Gene_ID\")\n",
    "    .merge(gene_effect_df, how='left', on=\"Metadata_NCBI_Gene_ID\")\n",
    ")\n",
    "\n",
    "orf_df.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Metadata_JCP2022                   7977\n",
       "Metadata_NCBI_Gene_ID              7977\n",
       "Metadata_Symbol                    7977\n",
       "Metadata_Approved_Symbol           7975\n",
       "Metadata_complexname               2242\n",
       "Metadata_protein_class             7955\n",
       "Metadata_disease_involvement       3470\n",
       "Metadata_subcellular_location      5712\n",
       "Metadata_gene_group_id             6857\n",
       "Metadata_homologue                 7704\n",
       "Metadata_biological_process        7674\n",
       "Metadata_cellular_component        6239\n",
       "Metadata_molecular_function        7648\n",
       "Metadata_wikipathway               4784\n",
       "Metadata_dependency_probability    7778\n",
       "Metadata_TPM                       7956\n",
       "Metadata_gene_effect               7776\n",
       "dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crispr_df = (\n",
    "    crispr_df.merge(corum_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(class_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(disease_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(location_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(gene_family, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(homolog_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(go_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(wp_df, how=\"left\", on=\"Metadata_Approved_Symbol\")\n",
    "    .merge(dependency_df, how=\"left\", on=\"Metadata_NCBI_Gene_ID\")\n",
    "    .merge(expression_df, how=\"left\", on=\"Metadata_NCBI_Gene_ID\")\n",
    "    .merge(gene_effect_df, how='left', on=\"Metadata_NCBI_Gene_ID\")\n",
    ")\n",
    "\n",
    "crispr_df.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "orf_df.to_csv('output/orf_metadata.tsv.gz', sep='\\t', index=False)\n",
    "crispr_df.to_csv('output/crispr_metadata.tsv.gz', sep='\\t', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "annotations",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
